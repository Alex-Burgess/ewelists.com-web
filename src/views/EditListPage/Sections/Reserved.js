import React, { useState, useEffect } from 'react';
// nodejs library to set properties for components
import PropTypes from "prop-types";
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
// @material-ui icons
// core components
import Table from "components/Table/Table.js";
import Card from "components/Card/Card.js";
import CardBody from "components/Card/CardBody.js";

import styles from "assets/jss/custom/views/editListPage/reservedStyle.js";
const useStyles = makeStyles(styles);

export default function SectionProducts(props) {
  const classes = useStyles();
  const { reserved, products } = props;
  const [desktop, setDesktop] = useState(true);

  useEffect( () => {
    function updateDimensions() {
      if (window.innerWidth < 600){
        setDesktop(false);
      } else {
        setDesktop(true);
      }
    };

    window.addEventListener('resize', updateDimensions);
    updateDimensions();
  }, []);

  const renderDesktopView = () => {
    return (
      <Table
        tableHead={[
          "",
          "ITEM",
          "QTY",
          "RESERVED BY"
        ]}
        tableData={
          renderDesktopTable()
        }
        tableShopping
        customHeadCellClasses={[
          classes.textCenter,
          classes.textCenter,
        ]}
        customHeadClassesForCells={[2, 3]}
        customCellClasses={[
          classes.tdName,
          classes.tdNumber + " " + classes.textCenter,
          classes.textCenter + " " + classes.tdName
        ]}
        customClassesForCells={[1, 2, 3, 4]}
      />
    )
  }

  const renderMobileView = () => {
    return (
      <Table
        tableHead={[
          "", ""
        ]}
        tableData={
          renderMobileTable()
        }
        tableShopping
      />
    )
  }

  const renderDesktopTable = () => {
    const allRows = reserved.map(
      (row, i) =>
            renderDesktopRow(products[row['productId']].productUrl, products[row['productId']].imageUrl, products[row['productId']].brand, products[row['productId']].details, row['name'], row['message'], row['quantity'])
    )

    allRows[reserved.length] =
      {
        addnew: true,
        colspan: "1",
        col: {
          colspan: 3,
        }
      }

    return allRows
  }

  const renderMobileTable = () => {
    const allRows = reserved.map(
      (row, i) =>
            renderMobileRow(products[row['productId']].productUrl, products[row['productId']].imageUrl, products[row['productId']].brand, products[row['productId']].details, row['name'], row['message'], row['quantity'])
    )

    allRows[reserved.length] =
      {
        addnew: true,
        colspan: "1",
        col: {
          colspan: 1,
        }
      }

    return allRows
  }

  const renderDesktopRow = (productUrl, imageUrl, brand, details, userName, message, quantity) => {
    return (
      [
      <div className={classes.imgContainer} key={1}>
        <a href={productUrl} target="_blank" rel="noopener noreferrer">
          <img src={imageUrl} alt="..." className={classes.img} />
        </a>
      </div>,
      <span key={1}>
        <a href={productUrl} target="_blank" rel="noopener noreferrer" className={classes.tdNameAnchor}>
          {brand}
        </a>
        <br />
        <small className={classes.tdNameSmall}>
          {details}
        </small>
      </span>,
      <span>
        {quantity}
      </span>,
      <span>
        {userName}
        <br />
        <small className={classes.tdNameSmall}>
          {message}
        </small>
      </span>
      ]
    )
  }

  const renderMobileRow = (productUrl, imageUrl, brand, details, userName, message, quantity) => {
    return (
      [
          <div className={classes.textCenter}>
            <div className={classes.imgContainer} key={1}>
              <a href={productUrl} target="_blank" rel="noopener noreferrer">
                <img src={imageUrl} alt="..." className={classes.img} />
              </a>
            </div>
            <a href={productUrl} target="_blank" rel="noopener noreferrer" className={classes.tdNameAnchor}>
              {brand}
            </a>
            <br />
            <small className={classes.tdNameSmall}>
              {details}
            </small>
          </div>,
          <div className={classes.textCenter}>
            <h4 className={classes.cardTitle}>
              {userName}
            </h4>
            <div className={classes.messageContainer}>
              <small className={classes.mobileMessage}>
                {message}
              </small>
            </div>
            <small className={classes.quantities}>
              Quantity: {quantity}
            </small>
          </div>
      ]
    )
  }

  return (
    <div className={classes.section}>
      <div className={classes.container}>
        <Card plain>
          <CardBody plain>
            {
              desktop
              ? renderDesktopView()
              : renderMobileView()
            }
          </CardBody>
        </Card>
      </div>
    </div>
  );
}

SectionProducts.propTypes = {
  reserved: PropTypes.array,
  products: PropTypes.object
};
