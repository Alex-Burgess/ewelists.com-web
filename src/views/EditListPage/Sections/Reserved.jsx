/*!

=========================================================
* Material Kit PRO React - v1.7.0
=========================================================

* Product Page: https://www.creative-tim.com/product/material-kit-pro-react
* Copyright 2019 Creative Tim (https://www.creative-tim.com)

* Coded by Creative Tim

=========================================================

* The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.

*/
import React from "react";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// @material-ui/core components
import withStyles from "@material-ui/core/styles/withStyles";
// @material-ui icons
// core components
import Table from "components/Table/Table.jsx";
import Card from "components/Card/Card.jsx";
import CardBody from "components/Card/CardBody.jsx";

import styles from "assets/jss/custom/views/editListPage/reservedStyle.jsx";


class SectionProducts extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      edit: {},
      width: window.innerWidth,
      height: window.innerHeight
    };
  }

  updateDimensions = () => {
    this.setState({ width: window.innerWidth, height: window.innerHeight });

    if (window.innerWidth < 600){
      this.setState({ desktop: false });
    } else {
      this.setState({ desktop: true });
    }
  };

  componentWillMount() {
    if (window.innerWidth < 600){
      this.setState({ desktop: false });
    } else {
      this.setState({ desktop: true });
    }
  }

  componentDidMount() {
    window.addEventListener('resize', this.updateDimensions);
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.updateDimensions);
  }

  handleEditClose(modal) {
    var x = [];
    x[modal] = false;
    this.setState({ edit: x });
  }

  handleEditClickOpen(modal) {
    console.log("Opening product id: " + modal)
    var x = [];
    x[modal] = true;
    this.setState({ edit: x });
  }

  renderDesktopView(classes, reserved) {
    return (
      <Table
        tableHead={[
          "",
          "NAME",
          "QTY",
          "MESSAGE"
        ]}
        tableData={
          this.renderDesktopTable(classes, reserved)
        }
        tableShopping
        customHeadCellClasses={[
          classes.textCenter,
          classes.textCenter,
        ]}
        customHeadClassesForCells={[2, 3]}
        customCellClasses={[
          classes.tdNumber + " " + classes.textCenter,
          classes.textCenter + " " + classes.tdMessage
        ]}
        customClassesForCells={[2, 3]}
      />
    )
  }

  renderMobileView(classes, reserved) {
    return (
      <Table
        tableHead={[
          "", ""
        ]}
        tableData={
          this.renderMobileTable(classes, reserved)
        }
        tableShopping
      />
    )
  }

  renderDesktopTable(classes, reserved: Reserved[]) {
    const allRows = reserved.map(
      (row, i) =>
            this.renderDesktopRow(classes, this.props.products[row['productId']].productUrl, this.props.products[row['productId']].imageUrl, row['name'], row['message'], row['quantity'])
    )

    allRows[reserved.length] =
      {
        addnew: true,
        colspan: "1",
        col: {
          colspan: 3,
        }
      }

    return allRows
  }

  renderMobileTable(classes, reserved: Reserved[]) {
    const allRows = reserved.map(
      (row, i) =>
            this.renderMobileRow(classes, this.props.products[row['productId']].productUrl, this.props.products[row['productId']].imageUrl, row['name'], row['message'], row['quantity'])
    )

    allRows[reserved.length] =
      {
        addnew: true,
        colspan: "1",
        col: {
          colspan: 1,
        }
      }

    return allRows
  }

  renderDesktopRow(classes, productUrl, imageUrl, userName, message, quantity) {
    return (
      [
      <div className={classes.imgContainer} key={1}>
        <a href={productUrl} target="_blank" rel="noopener noreferrer">
          <img src={imageUrl} alt="..." className={classes.img} />
        </a>
      </div>,
      <span>
        {userName}
      </span>,
      <span>
        {quantity}
      </span>,
      <span>
        {message}
      </span>
      ]
    )
  }

  renderMobileRow(classes, productUrl, imageUrl, userName, message, quantity) {
    return (
      [
          <div className={classes.textCenter}>
            <div className={classes.imgContainer} key={1}>
              <a href={productUrl} target="_blank" rel="noopener noreferrer">
                <img src={imageUrl} alt="..." className={classes.img} />
              </a>
            </div>
            <small className={classes.quantities}>
              Quantity: {quantity}
            </small>
          </div>,
          <div className={classes.textCenter}>
            <h4 className={classes.cardTitle}>
              {userName}
            </h4>
            <small className={classes.mobileDescription}>
              {message}
            </small>
          </div>
      ]
    )
  }

  render() {
    const { classes, reserved } = this.props;

    return (
      <div className={classes.section}>
        <div className={classes.container}>
          <Card plain>
            <CardBody plain>
              {
                this.state.desktop
                ? this.renderDesktopView(classes, reserved)
                : this.renderMobileView(classes, reserved)
              }
            </CardBody>
          </Card>
        </div>
      </div>
    );
  }
}

SectionProducts.propTypes = {
  classes: PropTypes.object,
  reserved: PropTypes.array,
  products: PropTypes.object
};

export default withStyles(styles)(SectionProducts);
