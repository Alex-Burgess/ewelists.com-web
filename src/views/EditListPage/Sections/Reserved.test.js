import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import TestRenderer from 'react-test-renderer';
import {mount} from 'enzyme';
import Reserved from './Reserved';

describe('Tests Reserved component of edit page.', () => {
  test('Renders desktop reserved table', async () => {
    let component;

    component = TestRenderer.create(
      <Router>
        <Reserved
          reserved={[
            {"productId":"12345678-prod-0003-1234-abcdefghijkl","quantity":1,"name":"John Smith","userId":"6c9b0a41-9a92-490c-98f8-c51f280c1234"},
            {"productId":"6a1f50a3-acb4-4842-8af1-5f606fed2ca7","quantity":2,"name":"John Smith","userId":"6c9b0a41-9a92-490c-98f8-c51f280c1234"}
          ]}
          products={{
            "12345678-prod-0003-1234-abcdefghijkl":{"productId":"12345678-prod-0003-1234-abcdefghijkl","quantity":1,"reserved":1,"type":"products","brand":"BABYZEN","details":"YOYO+ Puschair, Black with Aqua","productUrl":"https://www.johnlewis.com/babyzen-yoyo-pushchair-white-aqua/p4145291","imageUrl":"https://johnlewis.scene7.com/is/image/JohnLewis/237457570?$rsp-pdp-port-640$"},
            "6a1f50a3-acb4-4842-8af1-5f606fed2ca7":{"productId":"6a1f50a3-acb4-4842-8af1-5f606fed2ca7","quantity":4,"reserved":2,"type":"notfound","brand":"Test","details":"Some details","productUrl":"http://alink","imageUrl":"https://test.ewelists.com/images/product-default.jpg"}
          }}
        />
      </Router>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Renders mobile reserved table', async () => {
    let component;

    global.innerWidth = 375;

    component = TestRenderer.create(
      <Router>
        <Reserved
          reserved={[
            {"productId":"12345678-prod-0003-1234-abcdefghijkl","quantity":1,"name":"John Smith","userId":"6c9b0a41-9a92-490c-98f8-c51f280c1234"},
            {"productId":"6a1f50a3-acb4-4842-8af1-5f606fed2ca7","quantity":2,"name":"John Smith","userId":"6c9b0a41-9a92-490c-98f8-c51f280c1234"}
          ]}
          products={{
            "12345678-prod-0003-1234-abcdefghijkl":{"productId":"12345678-prod-0003-1234-abcdefghijkl","quantity":1,"reserved":1,"type":"products","brand":"BABYZEN","details":"YOYO+ Puschair, Black with Aqua","productUrl":"https://www.johnlewis.com/babyzen-yoyo-pushchair-white-aqua/p4145291","imageUrl":"https://johnlewis.scene7.com/is/image/JohnLewis/237457570?$rsp-pdp-port-640$"},
            "6a1f50a3-acb4-4842-8af1-5f606fed2ca7":{"productId":"6a1f50a3-acb4-4842-8af1-5f606fed2ca7","quantity":4,"reserved":2,"type":"notfound","brand":"Test","details":"Some details","productUrl":"http://alink","imageUrl":"https://test.ewelists.com/images/product-default.jpg"}
          }}
        />
      </Router>
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
