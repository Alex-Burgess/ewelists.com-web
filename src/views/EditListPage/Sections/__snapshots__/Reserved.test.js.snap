// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests Reserved component of edit page. Renders desktop reserved table 1`] = `
<div>
  <div
    className="makeStyles-container-1"
  >
    <div
      className="makeStyles-card-16 makeStyles-cardPlain-17"
    >
      <div
        className="makeStyles-cardBody-31 makeStyles-cardBodyPlain-33"
      >
        <div
          className="makeStyles-tableResponsive-52"
        >
          <table
            className="MuiTable-root makeStyles-table-46"
          >
            <thead
              className="MuiTableHead-root makeStyles-gray-44"
            >
              <tr
                className="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head makeStyles-tableHeadCell-49 makeStyles-tableCell-48 makeStyles-tableShoppingHead-47"
                  scope="col"
                >
                  
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head makeStyles-tableHeadCell-49 makeStyles-tableCell-48 makeStyles-tableShoppingHead-47"
                  scope="col"
                >
                  ITEM
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head makeStyles-tableHeadCell-49 makeStyles-tableCell-48 makeStyles-textCenter-4 makeStyles-tableShoppingHead-47"
                  scope="col"
                >
                  QTY
                </th>
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head makeStyles-tableHeadCell-49 makeStyles-tableCell-48 makeStyles-textCenter-4 makeStyles-tableShoppingHead-47"
                  scope="col"
                >
                  RESERVED BY
                </th>
              </tr>
            </thead>
            <tbody
              className="MuiTableBody-root"
            >
              <tr
                className="MuiTableRow-root undefined "
              >
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body makeStyles-tableCell-48 "
                >
                  <div
                    className="makeStyles-imgContainer-3"
                  >
                    <a
                      href="https://www.johnlewis.com/babyzen-yoyo-pushchair-white-aqua/p4145291"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      <img
                        alt="..."
                        src="https://johnlewis.scene7.com/is/image/JohnLewis/237457570?$rsp-pdp-port-640$"
                      />
                    </a>
                  </div>
                </td>
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body makeStyles-tableCell-48 makeStyles-tdName-7"
                >
                  <span>
                    <a
                      className="makeStyles-tdNameAnchor-8"
                      href="https://www.johnlewis.com/babyzen-yoyo-pushchair-white-aqua/p4145291"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      BABYZEN
                    </a>
                    <br />
                    <small
                      className="makeStyles-tdNameSmall-9"
                    >
                      YOYO+ Puschair, Black with Aqua
                    </small>
                  </span>
                </td>
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body makeStyles-tableCell-48 makeStyles-tdNumber-10 makeStyles-textCenter-4"
                >
                  <span>
                    1
                  </span>
                </td>
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body makeStyles-tableCell-48 makeStyles-textCenter-4 makeStyles-tdName-7"
                >
                  <span>
                    John Smith
                    <br />
                    <small
                      className="makeStyles-tdNameSmall-9"
                    />
                  </span>
                </td>
              </tr>
              <tr
                className="MuiTableRow-root undefined "
              >
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body makeStyles-tableCell-48 "
                >
                  <div
                    className="makeStyles-imgContainer-3"
                  >
                    <a
                      href="http://alink"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      <img
                        alt="..."
                        src="https://test.ewelists.com/images/product-default.jpg"
                      />
                    </a>
                  </div>
                </td>
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body makeStyles-tableCell-48 makeStyles-tdName-7"
                >
                  <span>
                    <a
                      className="makeStyles-tdNameAnchor-8"
                      href="http://alink"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Test
                    </a>
                    <br />
                    <small
                      className="makeStyles-tdNameSmall-9"
                    >
                      Some details
                    </small>
                  </span>
                </td>
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body makeStyles-tableCell-48 makeStyles-tdNumber-10 makeStyles-textCenter-4"
                >
                  <span>
                    2
                  </span>
                </td>
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body makeStyles-tableCell-48 makeStyles-textCenter-4 makeStyles-tdName-7"
                >
                  <span>
                    John Smith
                    <br />
                    <small
                      className="makeStyles-tdNameSmall-9"
                    />
                  </span>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Tests Reserved component of edit page. Renders mobile reserved table 1`] = `
<div>
  <div
    className="makeStyles-container-1"
  >
    <div
      className="makeStyles-card-16 makeStyles-cardPlain-17"
    >
      <div
        className="makeStyles-cardBody-31 makeStyles-cardBodyPlain-33"
      >
        <div
          className="makeStyles-tableResponsive-52"
        >
          <table
            className="MuiTable-root makeStyles-table-46"
          >
            <thead
              className="MuiTableHead-root makeStyles-gray-44"
            >
              <tr
                className="MuiTableRow-root MuiTableRow-head"
              >
                <th
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-head makeStyles-tableHeadCell-49 makeStyles-tableCell-48 makeStyles-textCenter-4 makeStyles-tableShoppingHead-47"
                  scope="col"
                >
                  RESERVED ITEMS
                </th>
              </tr>
            </thead>
            <tbody
              className="MuiTableBody-root"
            >
              <tr
                className="MuiTableRow-root undefined "
              >
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body makeStyles-tableCell-48 "
                >
                  <div
                    className="makeStyles-textCenter-4"
                  >
                    <div
                      className="makeStyles-imgContainer-3"
                    >
                      <a
                        href="https://www.johnlewis.com/babyzen-yoyo-pushchair-white-aqua/p4145291"
                        rel="noopener noreferrer"
                        target="_blank"
                      >
                        <img
                          alt="..."
                          src="https://johnlewis.scene7.com/is/image/JohnLewis/237457570?$rsp-pdp-port-640$"
                        />
                      </a>
                    </div>
                    <a
                      className="makeStyles-tdNameAnchor-8"
                      href="https://www.johnlewis.com/babyzen-yoyo-pushchair-white-aqua/p4145291"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      BABYZEN
                    </a>
                    <br />
                    <small
                      className="makeStyles-tdNameSmall-9"
                    >
                      YOYO+ Puschair, Black with Aqua
                    </small>
                  </div>
                </td>
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body makeStyles-tableCell-48 "
                >
                  <div
                    className="makeStyles-textCenter-4"
                  >
                    <h4
                      className="makeStyles-cardTitle-2"
                    >
                      John Smith
                    </h4>
                    <div
                      className="makeStyles-messageContainer-15"
                    >
                      <small
                        className="makeStyles-mobileMessage-13"
                      />
                    </div>
                    <small
                      className="makeStyles-quantities-14"
                    >
                      Quantity: 
                      1
                    </small>
                  </div>
                </td>
              </tr>
              <tr
                className="MuiTableRow-root undefined "
              >
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body makeStyles-tableCell-48 "
                >
                  <div
                    className="makeStyles-textCenter-4"
                  >
                    <div
                      className="makeStyles-imgContainer-3"
                    >
                      <a
                        href="http://alink"
                        rel="noopener noreferrer"
                        target="_blank"
                      >
                        <img
                          alt="..."
                          src="https://test.ewelists.com/images/product-default.jpg"
                        />
                      </a>
                    </div>
                    <a
                      className="makeStyles-tdNameAnchor-8"
                      href="http://alink"
                      rel="noopener noreferrer"
                      target="_blank"
                    >
                      Test
                    </a>
                    <br />
                    <small
                      className="makeStyles-tdNameSmall-9"
                    >
                      Some details
                    </small>
                  </div>
                </td>
                <td
                  aria-sort={null}
                  className="MuiTableCell-root MuiTableCell-body makeStyles-tableCell-48 "
                >
                  <div
                    className="makeStyles-textCenter-4"
                  >
                    <h4
                      className="makeStyles-cardTitle-2"
                    >
                      John Smith
                    </h4>
                    <div
                      className="makeStyles-messageContainer-15"
                    >
                      <small
                        className="makeStyles-mobileMessage-13"
                      />
                    </div>
                    <small
                      className="makeStyles-quantities-14"
                    >
                      Quantity: 
                      2
                    </small>
                  </div>
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</div>
`;
